lobby/src/test/java/org/triplea/lobby/server/login/LobbyLoginValidatorTest.java

```diff
@@ -104,50 +105,62 @@ final String md5Crypt(final String password) {
     }
 
     final void givenAnonymousAuthenticationWillFail() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.doesUserExist(user.getUsername())).thenReturn(true);
     }
 
     final void givenAnonymousAuthenticationWillSucceed() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.doesUserExist(user.getUsername())).thenReturn(false);
     }
 
     final void givenAuthenticationWillUseMd5CryptedPasswordAndSucceed() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.login(user.getUsername(), new HashedPassword(md5Crypt(PASSWORD)))).thenReturn(true);
     }
 
     final void givenAuthenticationWillUseMd5CryptedPasswordAndFail() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.login(user.getUsername(), new HashedPassword(md5Crypt(PASSWORD)))).thenReturn(false);
     }
 
     final void givenAuthenticationWillUseObfuscatedPasswordAndSucceed() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.login(user.getUsername(), new HashedPassword(obfuscate(PASSWORD)))).thenReturn(true);
     }
 
     final void givenAuthenticationWillUseObfuscatedPasswordAndFail() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.login(user.getUsername(), new HashedPassword(obfuscate(PASSWORD)))).thenReturn(false);
     }
 
     final void givenUserDoesNotExist() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.doesUserExist(user.getUsername())).thenReturn(false);
     }
 
     final void givenUserDoesNotHaveBcryptedPassword() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.getPassword(user.getUsername())).thenReturn(new HashedPassword(md5Crypt(PASSWORD)));
     }
 
     final void givenUserDoesNotHaveMd5CryptedPassword() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.getLegacyPassword(user.getUsername())).thenReturn(new HashedPassword(""));
     }
 
     final void givenUserExists() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.getUserByName(user.getUsername())).thenReturn(dbUser);
     }
 
     final void givenUserHasBcryptedPassword() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
       when(userDao.getPassword(user.getUsername())).thenReturn(new HashedPassword(bcrypt(PASSWORD)));
     }
 
     final void givenUserHasMd5CryptedPassword() {
+      when(databaseDao.getUserDao()).thenReturn(userDao);
```

Comment:
I'm not sure. I do know there is going to be more change landing to LobbyLoginValidator, so I'm not really interested to redesign the test case in its current form. The tests are going to be mostly re-written, I was tempted to do that directly in this PR but it was way too much. The updates to test code are really just what it takes to get them working, the tests I expect to be updated quite a lot. So it's pointless to redesign something that is going to be broken apart later and need redesign again.

Reply:
```
(Or you could add that to the existing @BeforeEach in case that works)
```

Perhaps not as unnecessary stubbings cause failures, it could be a huge PITA if a test case were added that did not require it.